README.md

```diff
@@ -64,17 +62,15 @@ semantics are applied.
 
 ### Interaction between `package.json` and `.npmignore` rules
 
-For simplicity, it is best to use _either_ a `files` list in `package.json`
-_or_ a `.npmignore` file, and not both.  If you only use one of these
-methods, you can skip this documentation section.
+In previous versions of this library, the `files` list in `package.json`
+was used as an initial filter to drive further tree walking. That is no
+longer the case as of version 6.0.0.
 
-The `files` list in `package.json` is used to direct the exploration of the
-tree.  In other words, that's all the walker will ever look at when
-exploring that level.
+If you have a `package.json` file with a `files` array within, any top
+level `.npmignore` and `.gitignore` files *will be ignored*.
 
-In some cases this can lead to a `.npmignore` file being ignored.  If a
-_directory_ is listed in `files`, then any rules in a root or nested
-`.npmignore` files will be honored.
+If a _directory_ is listed in `files`, then any rules in nested `.npmignore`
+files will be honored.
```

Comment:
nested npmignore files _within that directory_?

Reply:
right. meaning if you add `bin` to your `files` array, and the contents of `bin` are `bin/foo` and `bin/bar` but a `bin/.npmignore` file exists and lists `./foo` then `bin/foo` will not be published

Reply:
Makes sense - may i suggest:
```suggestion
If a _directory_ is listed in `files`, then any rules in nested `.npmignore` files within that directory will be honored.
```